libpod/container_api.go

```diff
@@ -478,13 +479,15 @@ func (c *Container) RemoveArtifact(name string) error {
 }
 
 // Wait blocks until the container exits and returns its exit code.
-func (c *Container) Wait() (int32, error) {
-	return c.WaitWithInterval(DefaultWaitInterval)
+func (c *Container) Wait(ctx context.Context) (int32, error) {
```

Comment:
I made the wait functions cancel-able. Useful for not leaking gorutines when client cancels request (closes connection abruptly). For CLI podman it's not used since context passed here is not canceled on SIGTERM or SIGINT. CLI chooses IMO rather brutal `os.Exit()`.